src/store/getters/state-getters.ts

```diff
@@ -727,18 +724,24 @@ export const hasRecDeliveryChanged = (state: StateIF): boolean => {
 
 /** True if any people/roles have changed. */
 export const havePeopleAndRolesChanged = (state: StateIF): boolean => {
-  const currentOrgPersons = getOrgPeople(state)?.map(op => {
-    const isOrg = (op.officer.partyType === PartyTypes.ORGANIZATION)
-
-    // add fields that are not in the snapshot
-    return {
-      deliveryAddress: { deliveryInstructions: null, streetAddressAdditional: null, ...op.deliveryAddress },
-      mailingAddress: { deliveryInstructions: null, streetAddressAdditional: null, ...op.mailingAddress },
-      officer: isOrg ? { identifier: null, taxId: null, ...op.officer } : { ...op.officer },
-      roles: [ ...op.roles ]
+  /** Normalizes fields that may be empty, null or undefined. */
+  function normalize (op: OrgPersonIF): OrgPersonIF {
+    if (!op.deliveryAddress.deliveryInstructions) op.deliveryAddress.deliveryInstructions = ''
+    if (!op.deliveryAddress.streetAddressAdditional) op.deliveryAddress.streetAddressAdditional = ''
+    if (!op.mailingAddress.deliveryInstructions) op.mailingAddress.deliveryInstructions = ''
+    if (!op.mailingAddress.streetAddressAdditional) op.mailingAddress.streetAddressAdditional = ''
+    if (!op.officer.email) op.officer.email = null
+
+    if (op.officer.partyType === PartyTypes.ORGANIZATION) {
+      if (!op.officer.identifier) op.officer.identifier = null
+      if (!op.officer.taxId) op.officer.taxId = null
     }
-  })
-  const originalOrgPersons = getEntitySnapshot(state)?.orgPersons
+
+    return op
+  }
```

Comment:
Depending on how each record was initialized or stored in the db, these properties may be '' or null or undefined. This normalizes them so the comparison on line 746 works in all cases.

(When the comparison failed, various filings identified that the People and Roles changed but they really hadn't.)